content/articles/json-server-and-axios/index.md

```diff
@@ -0,0 +1,422 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /json-server-and-axios/
+title: Building a Vue.js 3 application with JSON server and Axios
+description: This tutorial will help the learner be more confident in their ability to interface with APIs using Axios and Vue 3.
+author: terrypha-wamaitha
+date: 2021-12-29T00:00:00-21:50
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/json-server-and-axios/hero.png
+    alt: Building a Vue.js 3 application with JSON server and Axios
+---
+
+Vue is a front-end JavaScript framework used to create single-page apps that run on the client.
+<!--more-->
+It can also be used to create full-stack applications by making HTTP requests to a backend server. It is popular with Node.js and Express(MEVN stack).
+
+[JSON server](https://www.npmjs.com/package/json-server) is an npm package that lets you create mock REST APIs with zero coding. It is used to create a simple JSON file that can be used as a database and responds to HTTP requests.
+
+Axios is a HTTP client to make HTTP requests to the [JSON server](https://www.npmjs.com/package/json-server).
+
+In this article, we will build a shopping list application. We will start with a blank Vue.js application then add `json-server` for local data storage, and Axios for making HTTP requests:
+
+![shopping-list](/engineering-education/json-server-and-axios/shopping-list.png)
+
+### Table of contents
+- [New Vue.js application](#new-vuejs-application)
+- [Creating a JSON file and installing JSON-Server](#creating-a-json-file-and-installing-json-server)
+- [Getting HTTP data within our application](#getting-http-data-within-our-application)
+- [Adding data using POST](#adding-data-using-post)
+- [Updating data using PATCH](#updating-data-using-patch)
+- [Deleting data using DELETE](#deleting-data-using-delete)
+
+### Prerequisites
+To follow along with this tutorial, you are required to have the following:
+- [Node.js](https://nodejs.org/en/download/package-manager/) installed
+- Some knowledge of JavaScript
+
+### New Vue.js application
+Assuming you have installed [Node.js](https://nodejs.org/en/download/package-manager/) in your computer, run the following commands in your terminal to install Vue CLI:
+
+```bash
+#Install Vue.js CLI globally
+$ npm install -g @vue/cli
+```
+
+After the installation, create a new project using:
+
+```bash
+$ vue create shopping-list 
+```
+
+Navigate to the folder and open the project in your favorite editor. e.g VS Code:
+
+```bash
+$ cd shopping-list
+
+# open project in vs code
+$ code .
+```
+
+Run the following command to start the Vue.js application:
+
+```bash
+$ npm run serve
+```
+
+Our application will run on `http://localhost:8080/` in the browser. We should now see a default homepage of Vue running on our browser.
+
```

Comment:
Write the application link as a link. ie. like this:

<http://localhost:8080/>